### PR TITLE
[e2e] Refactor of e2e/framework/authorizer_util.go

### PR DESCRIPTION
**Pull request by johnSchnake**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
- moves these helper functions into e2e/framework/auth
 - removes logging from helper functions
 - in some cases explicitly returns errors that were implicitly
ignored/logged. In the situations where they should be ignored,
we explicitly check that the condition is met before ignoring it.
 - fixes references of these methods to use the right package and
return values.

Ref #76206 (parent issue #75601 )

/area test
/area test-framework
/sig testing

### CODE REVIEW

File: test/e2e/framework/auth/helpers.go
Comment (johnSchnake) [inline test/e2e/framework/auth/helpers.go]: If we only document this err path as being for GKE lets test it as such...

File: test/e2e/framework/auth/helpers.go
Comment (johnSchnake) [inline test/e2e/framework/auth/helpers.go]: If we only want to hide the error because "RBAC may simply be disabled" lets just test it as such.

File: test/e2e/framework/auth/helpers.go
Comment (johnSchnake) [inline test/e2e/framework/auth/helpers.go]: We've filtered out the RBAC disabled case; now it is a real error we should return to the caller.

File: test/e2e/auth/pod_security_policy.go
Comment (johnSchnake) [inline test/e2e/auth/pod_security_policy.go]: goimports does this now for performance reasons since the path uses a symlink and the import path isnt the real path.

File: test/e2e/framework/auth/helpers.go
Comment (andrewsykim) [inline test/e2e/framework/auth/helpers.go]: Any reason for not using `IsRBACEnabled(f)` instead? Seems like there's potential for race conditions otherwise

File: test/e2e/auth/pod_security_policy.go
Comment (andrewsykim) [reply test/e2e/auth/pod_security_policy.go]: Ah, this is good to know. Always thought it was a vscode thing

Comment (johnSchnake) [reply test/e2e/auth/pod_security_policy.go]: You're the 2nd person to suggest that to me but I tested manually with goimports and confirmed its not IDE based.

File: test/e2e/framework/auth/helpers.go
Comment (johnSchnake) [reply test/e2e/framework/auth/helpers.go]: So I didnt have `f` in this context.

Though now that you mention it, I really do need to just add that to the parameter list and update the references. For some reason I was assuming that `IsRBACEnabled` would have been called and so checking the variable would be fine since it was a `Once` type method. However, there is no guarantee it has been run at all, in which case the variable would be false even though RBAC is enabled.

Comment (johnSchnake) [reply test/e2e/framework/auth/helpers.go]: That method is actually just looking at the list of cluster roles so I'm going to take a look at it and see about modifying the signature. I don't need to rely on the entire framework to just inspect a list.

**Comment by johnSchnake** [comment]
Hitting an import cycle now because we check for the provider.

`ProviderIs` is in the `framework` package which causes an import loop.

Trying to push that function down into another package by itself fails because `ProviderIs` requires access to the `TestContext` global variable, which is also defined in `framework`.

Currently discussing the idea of refactoring the TestContext to be in its own package farther down in the stack so that other packages can depend on it without pulling in `framework` and causing a cycle.

(Initially proposed in #76206, waiting on that discussion to continue).

File: test/e2e/framework/auth/helpers.go
Comment (johnSchnake) [inline test/e2e/framework/auth/helpers.go]: This is the _only_ line that ties this package to the framework. Considering how hard it is to decouple the testContext/framework/etc at this time, I think we should just leave this as-is with a FIXME line.

File: test/e2e/framework/auth/helpers.go
Comment (johnSchnake) [reply test/e2e/framework/auth/helpers.go]: Unfortunately, this dependence on the `framework` was too hard to break so I removed this check and added a FIXME comment.

The `ProviderIs` depends on the `TestContext` which depends on the `Framework` object. Those are other, large items in the framework that we can consider breaking down/putting into their own packages but at this time it is not as feasible.

Removing that check isn't a regression at least, it is just returning the functionality to the way it was before.

File: test/e2e/framework/auth/helpers.go
Comment (johnSchnake) [inline test/e2e/framework/auth/helpers.go]: I ended up including some logging functionality in the IsRBACEnabled method; if you want me to add this back I can.

It is not an error condition based for tests and seems like it only happens in known situations that won't surprise anyone.

File: test/e2e/framework/auth/helpers.go
Comment (johnSchnake) [inline test/e2e/framework/auth/helpers.go]: Added the err to the message here; not sure why it wasn't included in the first place.

File: test/e2e/framework/auth/helpers.go
Comment (johnSchnake) [inline test/e2e/framework/auth/helpers.go]: Added the err to the message here; not sure why it wasn't included in the first place.

File: test/e2e/framework/auth/helpers.go
Comment (johnSchnake) [inline test/e2e/framework/auth/helpers.go]: Technically we went from the `rbackV1` to `v1beta1rbac` but it is consistent with the other types throughout. I wanted to allow the callers to just pass the one object `f.ClientSet.RbacV1beta1()` rather than having to require multiple parameters.

File: test/e2e/framework/auth/helpers.go
Comment (andrewsykim) [inline test/e2e/framework/auth/helpers.go]: As a follow-up we should bump these to v1

File: test/e2e/framework/auth/helpers.go
Comment (andrewsykim) [inline test/e2e/framework/auth/helpers.go]: As a follow-up, would putting all the log methods in `test/e2e/framework/log` help with the import issues you mentioned?

File: test/e2e/framework/auth/helpers.go
Comment (andrewsykim) [reply test/e2e/framework/auth/helpers.go]: I think we should leave the log line for now using logf below

File: test/e2e/framework/auth/helpers.go
Comment (andrewsykim) [reply test/e2e/framework/auth/helpers.go]: makes sense, seems like we should update the callers to v1 as well, but can be a follow-up

**Comment by johnSchnake** [comment]
The gce-storage-slow job failed with an _interesting_ error:
```
W0412 18:08:00.868] 2019/04/12 18:08:00 main.go:275: Flag parse failed: unknown flag: --gingko-parallel
```
But that flag was added 2 years ago :-/ I didnt change anything related to that. An odd type of error to get unexpectedly.

**Comment by johnSchnake** [comment]
I just realized the error there was a typo in the flag: `gingko-parallel` should be `ginkgo-parallel`. Going to figure out where that came from...

**Comment by andrewsykim** [comment]
Good catch, looks like it's from https://github.com/kubernetes/test-infra/pull/12094

**Comment by johnSchnake** [comment]
Yea, made an issue and submitted a patch. My first to test-infra :) https://github.com/kubernetes/test-infra/pull/12185

**Comment by johnSchnake** [comment]
Ugh, then my manually resolved conflict in the bazel files caused a failure because of the order. TIL, easy to avoid in the future.

**Comment by johnSchnake** [comment]
Rebased; all the refactors keep causing conflicts. Need lgtm again @andrewsykim

File: test/e2e/auth/audit.go
Comment (timothysc) [inline test/e2e/auth/audit.go]: Why do you need this change, it seems cosmetic.

File: test/e2e/framework/auth/helpers.go
Comment (timothysc) [inline test/e2e/framework/auth/helpers.go]: Open an issue and add a TODO() w/link

File: test/e2e/framework/auth/helpers.go
Comment (timothysc) [reply test/e2e/framework/auth/helpers.go]: Why not use errors.Wrapf then here.

File: test/e2e/framework/auth/helpers.go
Comment (timothysc) [reply test/e2e/framework/auth/helpers.go]: Please open an issue and link to the parent with a //TODO.

**Review by timothysc — CHANGES_REQUESTED** [review]
/lgtm cancel 
/approve 

Please address comments, then lgtm from me.

File: test/e2e/framework/auth/helpers.go
Comment (johnSchnake) [reply test/e2e/framework/auth/helpers.go]: You're right, I should have. Will fix.

File: test/e2e/auth/audit.go
Comment (johnSchnake) [reply test/e2e/auth/audit.go]: It is cosmetic but @andrewsykim requested it and I believe it is consistent with some of the other references to this package.

**Comment by johnSchnake** [comment]
Added `TODO` with link to issues and changed `fmt.Errorf` to the `errors.Wrapf` call.

Ready for another peek @timothysc

**Comment by johnSchnake** [comment]
Had to rebase due to conflict again...

File: test/e2e/auth/audit.go
Comment (timothysc) [reply test/e2e/auth/audit.go]: TL;DR when doing refactoring work, decouple these changes as a separate PR of commit so it's clear to the approver what the intent of your PR is.  Each commit should do just (1) thing.

**Review by timothysc — APPROVED** [review]
/lgtm - thx for the patch @johnSchnake

**Pull request merged**
